### DISCUSSION TITLE
myAction not working

### DISCUSSION THREAD

**Discussion by thelulzy** [elgatosf/elgato#127] [Help] [answered]
I'm trying to get started with developing my first streamdeck plugin and I'm finding that the `Stream Deck connected!` message from the template is outputting in the debug console, but the `Your key code goes here!` from the myAction.onKeyUp method isn't when pressing and releasing the `Example Action` button. 

***
*Edit: I've also noticed the `Message from external window.` console output from `external.js` works as expected.
*** 
*Edit 2: I **_do not_** get the yellow triangle w/ exclamation mark error icon when pressing/releasing the `Example Action` button.
***

Clearly I've messed something up from the [getting started guide](https://docs.elgato.com/sdk/plugins/getting-started), but I'm struggling to figure out what I've done wrong.

`app.js`:
```js
/// <reference path="libs/js/action.js" />
/// <reference path="libs/js/stream-deck.js" />

const myAction = new Action('com.lulz.windows-multimedia.sdPlugin');

/**
 * The first event fired when Stream Deck starts
 */
$SD.onConnected(({ actionInfo, appInfo, connection, messageType, port, uuid }) => {
	console.log('Stream Deck connected!');
});

myAction.onKeyUp(({ action, context, device, event, payload }) => {
	console.log('Your key code goes here!');
});

myAction.onDialRotate(({ action, context, device, event, payload }) => {
	console.log('Your dial code goes here!');
});
```

`manifest.json`:
```js
{
	"SDKVersion": 2,
	"Author": "Lulz",
	"CodePath": "app.html",
	"Description": "Windows multimedia control and info",
	"Name": "Stream Deck Windows Multimedia",
	"Icon": "actions/template/assets/action",
	"URL": "",
	"Version": "0.0.1",
	"Software": {
		"MinimumVersion": "5.0"
	},
	"OS": [
		{
			"Platform": "mac",
			"MinimumVersion": "10.11"
		},
		{
			"Platform": "windows",
			"MinimumVersion": "10"
		}
	],
	"Category": "Template Plugin",
	"CategoryIcon": "actions/template/assets/actionimage",
	"Actions": [
		{
			"Icon": "actions/template/assets/actionimage",
			"Name": "Example Action",
			"Controllers": ["Keypad", "Encoder"],
			"Encoder": {
				"layout": "$B1",
				"TriggerDescription": {
					"Rotate": "Describe the rotation",
					"Push": "Describe the encoder push",
					"Touch": "Describe the touch",
					"LongTouch": "Describe the long touch"
				}
			},
			"States": [
				{
					"Image": "actions/template/assets/action"
				}
			],
			"Tooltip": "This is an example tooltip",
			"UUID": "com.lulz.windows-multimedia.sdPlugin",
			"PropertyInspectorPath": "actions/template/property-inspector/inspector.html"
		}
	]
}
```

`File structure`:

![image](https://github.com/user-attachments/assets/60f513a9-2215-4c10-9166-8cb7d14c5751)

**Comment by ChekTek** [comment]
Hi @thelulzy,

Your code looks good to me! Does the libs folder include the contents of the [javascript sdk](https://github.com/elgatosf/streamdeck-javascript-sdk)?

It should not matter, but you may want to adjust your action's UUID to something that describes the action i.e. `com.lulz.windows-multimedia.next-song`.

If you are starting a new plugin, you may want to consider our new Node.js SDK! You can get started with the [CLI tool](https://www.npmjs.com/package/@elgato/cli), which streamlines a ton of the process and is much easier to work with (although the documentation is not available yet). 

Plugins created with the CLI tool run in the Node.js environment, which removes many of the limitations that were in place before when running an HTML file in the Qt Web-engine.

The CLI tool also provide a way to restart your plugin when you make changes, so you no longer have to completely close Stream Deck to register your code changes.

**Comment by soap-less** [comment, marked as answer]
Was running into the same issue, turned out that having a UUID with a capital letter was messing it up. i just set it all to lower case and it started working.
